File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        curMax=float('-inf')
        res=[-1]*len(arr)
        for i in range(len(arr)-2, -1, -1):
            curMax=max(curMax, arr[i+1])
            res[i]=curMax
        
        return res

File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int curMax=Integer.MIN_VALUE;
        int res[]=new int[arr.length];
        Arrays.fill(res, -1);
        for(int i=arr.length-2; i>=0; i--){
            curMax=Math.max(curMax, arr[i+1]);
            res[i]=curMax;
        }
        return res;
    }
}